zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/tenant/TenantControllerTest.java

```diff
@@ -0,0 +1,224 @@
+
+  @MockBean private TenantServices tenantServices;
+
+  @BeforeEach
+  void setup() {
+    when(tenantServices.withAuthentication(any(Authentication.class))).thenReturn(tenantServices);
+  }
+
+  @Test
+  void createTenantShouldReturnAccepted() {
+    // given
+    final var tenantName = "Test Tenant";
+    final var tenantId = "tenant-test-id";
+    when(tenantServices.createTenant(new TenantDTO(null, tenantId, tenantName)))
+        .thenReturn(
+            CompletableFuture.completedFuture(
+                new TenantRecord().setTenantKey(100L).setName(tenantName).setTenantId(tenantId)));
+
+    // when
+    webClient
+        .post()
+        .uri(TENANT_BASE_URL)
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(new TenantCreateRequest().name(tenantName).tenantId("tenant-test-id"))
+        .exchange()
+        .expectStatus()
+        .isAccepted();
+
+    // then
+    verify(tenantServices, times(1)).createTenant(new TenantDTO(null, tenantId, tenantName));
+  }
+
+  @Test
+  void createTenantWithEmptyTenantIdShouldFail() {
+    // given
+    final var tenantName = "Tenant Name";
+
+    // when
+    webClient
+        .post()
+        .uri(TENANT_BASE_URL)
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(new TenantCreateRequest().name(tenantName))
+        .exchange()
+        .expectStatus()
+        .isBadRequest()
+        .expectBody()
+        .json(
+            """
+            {
+              "type": "about:blank",
+              "status": 400,
+              "title": "INVALID_ARGUMENT",
+              "detail": "No tenantId provided.",
+              "instance": "%s"
+            }"""
+                .formatted(TENANT_BASE_URL));
+
+    // then
+    verifyNoInteractions(tenantServices);
+  }
+
+  @Test
+  void updateTenantShouldReturnUpdatedResponse() {
+    // given
+    final var tenantKey = 100L;
+    final var tenantName = "Updated Tenant Name";
+    final var tenantId = "tenant-test-id";
+    when(tenantServices.updateTenant(new TenantDTO(tenantKey, null, tenantName)))
+        .thenReturn(
+            CompletableFuture.completedFuture(
+                new TenantRecord()
+                    .setName(tenantName)
+                    .setTenantKey(tenantKey)
+                    .setTenantId(tenantId)));
+
+    // when
+    webClient
+        .patch()
+        .uri("%s/%s".formatted(TENANT_BASE_URL, tenantKey))
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(new TenantUpdateRequest().name(tenantName))
+        .exchange()
+        .expectStatus()
+        .isOk()
+        .expectBody()
+        .json(
+            """
+              {
+                "key": 100,
+                "tenantId": "tenant-test-id",
+                "name": "Updated Tenant Name"
+              }
+              """);
```

Comment:
We could use string formatting to inject id and name into the expected response here.